pkg/volume/glusterfs/glusterfs.go

```diff
@@ -671,6 +680,10 @@ func parseClassParameters(params map[string]string, kubeClient clientset.Interfa
 		return nil, fmt.Errorf("StorageClass for provisioner %s must contain 'resturl' parameter", glusterfsPluginName)
 	}
 
+	if cfg.gid != 0 {
+		cfg.gid = defaultGid
```

Comment:
Is that really what you want to do, @humblec ? Or do you rather mean if cfg.gid == 0 , then cfg.gid = defaultGid ? I'd say don't use a default gid, but if none was specified explicitly, then it's 0 (root).